src/plugins/data/public/ui/query_editor/query_editor.tsx

```diff
@@ -289,12 +293,29 @@ export default class QueryEditorUI extends Component<Props, State> {
     const headerClassName = classNames('osdQueryEditorHeader', this.props.headerClassName);
     const bannerClassName = classNames('osdQueryEditorBanner', this.props.bannerClassName);
 
+    const useQueryEditor =
+      this.props.query.language === 'SQLAsync' ||
```

Comment:
can add this boolean here: https://github.com/opensearch-project/OpenSearch-Dashboards/blob/96fb9d629e6539db4dacd922df3fff6809fdb407/src/plugins/data/public/ui/types.ts#L23